keps/sig-storage/3751-volume-attributes-class/README.md

```diff
@@ -175,24 +175,28 @@ type PersistentVolumeClaimSpec struct {
 // PersistentVolumeClaimStatus represents the status of PV claim
 type PersistentVolumeClaimStatus struct {
   ...
-  // The VolumeAttributesClassName string cannot be empty
-  VolumeAttributesClassName string
-  ModifyVolumeStatus *PersistentVolumeClaimModifyVolumeStatus
+  ModifyVolumeStatus ModifyVolumeStatus
   ...
 }
 
+type ModifyVolumeStatus struct {
+  ActualVolumeAttributesClassName string
+  TargetVolumeAttributesClassName string
+  Status *PersistentVolumeClaimModifyVolumeStatus
+}
+
 // +enum
 type PersistentVolumeClaimModifyVolumeStatus string
 
 const (
   // When modify volume is complete, the empty string is set by modify volume controller.
-  PersistentVolumeClaimNoModifyVolumeInProgress PersistentVolumeClaimModifyVolumeStatus = ""
+  PersistentVolumeClaimNoModifyVolumeInSucceeded PersistentVolumeClaimModifyVolumeStatus = ""
   // State set when modify volume controller starts modifying the volume in control-plane
+  // When the request has been rejected as invalid by the CSI driver. To resolve this error,
+  // the PersistentVolumeClaim needs to specify a valid VolumeAttributesClass.
   PersistentVolumeClaimControllerModifyVolumeInProgress PersistentVolumeClaimModifyVolumeStatus = "ControllerModifyVolumeInProgress"
   // State set when modify volume has failed in modify volume controller with a terminal error.
-  // Transient errors such as timeout should not set this status and should leave ModifyVolumeStatus
-  // unmodified, so as modify volume controller can resume the volume modification.
-  PersistentVolumeClaimControllerModifyVolumeFailed PersistentVolumeClaimModifyVolumeStatus = "ControllerModifyVolumeFailed"
+  PersistentVolumeClaimControllerModifyVolumeInfeasible PersistentVolumeClaimModifyVolumeStatus = "ControllerModifyVolumeInfeasible"
```

Comment:
How to reconginze whether an error is a retryable error or not? We need to clairfy it first. i.e. timeout error ?

Reply:
The description is in the paragraph above, this is duplicating. Right now a invalid argument error is an infeasible error. If you have more in mind please advice.

Reply:
So we don't need `Failed` anymore? What about other errors such as - may be volume has entered a state where it can't be modified right now, but can be modified like 10 minutes later (like a quota on request/hour or something for same volume).

Reply:
Discussed today at the CSI meeting. We will have Pending/InProgress/Infeasible/""(indicates success)